torch/csrc/distributed/c10d/NCCLUtils.hpp

```diff
@@ -320,14 +320,14 @@ class NCCLComm {
     comm->initialized_ = isInitialized;
     return comm;
   }
-#endif
```

Comment:
I'm curious, why did this `#endif` move down?

Reply:
`ncclConfig_t` is a new type introduced in nccl 2.14. So `static std::shared_ptr<NCCLComm> split()` also need to be wrapped by `NCCL_HAS_COMM_NONBLOCKING` aka nccl >= 2.14.